lib/semantics/type.h

```diff
@@ -251,6 +251,10 @@ class DerivedTypeSpec {
   bool operator==(const DerivedTypeSpec &that) const {
     return &typeSymbol_ == &that.typeSymbol_ && parameters_ == that.parameters_;
   }
+
+  // Do the kind type parameters of this have the same values as the
+  // corresponding kind type parameters of that?
+  bool IsKindCompatibleWith(const DerivedTypeSpec &) const;
```

Comment:
I'm curious why this functionality was moved into the class rather than exposing the ordered list of parameter decls. Of course, one could have both with a functional style `IsKindCompatible(x, y)` wrapper that just returns `x->IsKindCompat(y)`.